version: '3.2'
services:
  messaging:
    image: gcr.io/panoptes-survey/pocs-base
    privileged: true
    env_file: env_file
    volumes:
      - pandir:/var/panoptes
    command: bash -c "python3 $POCS/scripts/run_messaging_hub.py --from_config"
  weather:
    image: gcr.io/panoptes-survey/pocs-base
    privileged: true
    env_file: env_file
    depends_on:
      - "messaging"    
    volumes:
      - pandir:/var/panoptes
    command: bash -c "python3 $POCS/scripts/docker/run_weather.py"
  peas:
    image: gcr.io/panoptes-survey/pocs-base
    privileged: true
    env_file: env_file
    depends_on:
      - "messaging"    
    volumes:
      - pandir:/var/panoptes
    command: bash -c "python3 $POCS/scripts/docker/run_peas.py"    
  paws:
    image: gcr.io/panoptes-survey/paws
    privileged: true
    env_file: env_file
    depends_on:
      - "messaging"    
    ports:
      - target: 8888
        published: 8888
        protocol: tcp
        mode: host
    volumes:
      - pandir:/var/panoptes
  pocs:
    image: gcr.io/panoptes-survey/pocs-base
    privileged: true
    env_file: env_file
    depends_on:
      - "messaging"
    volumes:
      - pandir:/var/panoptes
    # No-op for now - connect via "docker exec -it panoptes_pocs_1 jupyter-console"
    command: tail -f /dev/null  
volumes:
    pandir:
      driver: local
      driver_opts:
        type: none
        device: /var/panoptes
        o: bind